### PR TITLE
(WIP) Add support for encrypting session state

### PR DESCRIPTION
**Pull request by totten**
Overview
----------------------------------------
_A brief description of the pull request. Keep technical jargon to a minimum. Hyperlink relevant discussions._

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by demeritcowboy** [comment]
jenkins test this please (`Unable to send e-mail. Contact the site administrator if the problem     [error]
persists.`)

**Comment by seamuslee001** [comment]
Jenkins re test this please

File: CRM/Utils/Cache/SerializationTrait.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  /**
+   * @param string $data
+   * @return mixed
+   * @see \unserialize()
+   */
+  protected function unserialize($data) {
+    if ($this->serializer === NULL) {
+      return unserialize($data);
```
Comment (seamuslee001) [inline CRM/Utils/Cache/SerializationTrait.php:64]: @totten do we need to run this through `xKerman\Restricted\unserialize` ?

**Comment by jaapjansma** [comment]
@totten or @seamuslee001 what is the status of this PR? It looks like it requires additional work. Are you planning to continue this work? Otherwise we will propose that we close this PR.

**Comment by eileenmcnaughton** [comment]
Agree with @jaapjansma - closing the PR doesn't delete the work but it's adding no value being open (nearly a year later)

**Pull request closed without merging**